### PR TITLE
regex: fix unset max_program_size handling

### PR DESCRIPTION
**Pull request by EricSesterhennX41**
This addresses https://oss-fuzz.com/testcase-detail/5787823586803712
When max_program_size is specified but not set it defaults to 0,
which this patch treats as an unlimited size. Otherwise an exception will be 
thrown by the attached testcase.

Risk Level: Low.
Testing: Corpus entry added
Docs Changes: N/A.
Release Notes: N/A.
Platform Specific Features: N/A.

Signed-off-by: Eric Sesterhenn <eric.sesterhenn@x41-dsec.de>

### CODE REVIEW

**Comment by daixiang0** [comment]
Seems need that case that max_program_size as 0.

File: source/common/common/regex.cc [hunk 1]
```diff
@@ -48,7 +48,7 @@ CompiledGoogleReMatcher::CompiledGoogleReMatcher(
   if (config.google_re2().has_max_program_size()) {
     const uint32_t max_program_size =
         PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.google_re2(), max_program_size, 100);
-    if (regex_program_size > max_program_size) {
+    if (max_program_size > 0 && regex_program_size > max_program_size) {
```
Comment (mattklein123) [inline source/common/common/regex.cc:51]: I could go either way on this, but I actually feel that the exception is correct. If someone has explicitly configured that setting but not specified a value, by proto rules, they have specified 0. I think the exception is valid? Thoughts?

/wait-any

**Review by yanavlasov — CHANGES_REQUESTED** [review]

**Comment by EricSesterhennX41** [comment]
I have prepared and tested a patch for this with the solution you suggested (it requires updating api/envoy/type/matcher/v3/regex.proto for this testcase). It will fix the testcase above by enforcing the size > 0, but the fuzzer might set it to 1 in another run and throw the exception as well. While testing i easily triggered the check for the default value of 100. Should I submit the patch to the proto files you suggested? At the moment I am not sure how to properly address this for all cases except by maybe catching the exception, but that does not feel correct either. Any suggestions?

**Pull request closed without merging**